samcli/commands/deploy/deploy_context.py

```diff
@@ -267,14 +267,8 @@ def deploy(
                     if not click.confirm(f"{self.MSG_CONFIRM_CHANGESET}", default=False):
                         return
 
-                # Stop the rollback in the case of DO_NOTHING or if this is a new stack and DELETE is specified
-                # DO_NOTHING behaves the same disable_rollback, they both preserve the current state of the stack
-                do_disable_rollback = (
-                    self.on_failure in [FailureMode.DO_NOTHING, FailureMode.DELETE] or disable_rollback
-                )
-
-                self.deployer.execute_changeset(result["Id"], stack_name, do_disable_rollback)
-                self.deployer.wait_for_execute(stack_name, changeset_type, do_disable_rollback, self.on_failure)
+                self.deployer.execute_changeset(result["Id"], stack_name, disable_rollback)
+                self.deployer.wait_for_execute(stack_name, changeset_type, disable_rollback, self.on_failure)
```

Comment:
From this my takeaway is we promote explicit behavior overall?

Reply:
The `--disable-rollback` option will now have to be explicitly defined for it to stop rollbacks in a change set like before, since this can affect resource replacements and we don't want to infer that from a different option.